tachyon/zk/plonk/circuit/floor_planner/allocated_region.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_ALLOCATED_REGION_H_
+#define TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_ALLOCATED_REGION_H_
+
+#include <stddef.h>
+
+#include "tachyon/export.h"
+
+namespace tachyon::zk {
+
+// A region allocated within a column.
+class TACHYON_EXPORT AllocatedRegion {
+ public:
+  constexpr AllocatedRegion(size_t start, size_t length)
+      : start_(start), length_(length) {}
+
+  constexpr size_t start() const { return start_; }
+  constexpr size_t length() const { return length_; }
+
+  constexpr bool operator<(const AllocatedRegion& other) const {
+    return start_ < other.start_;
+  }
```

Comment:
Here is original code.
```
impl Ord for AllocatedRegion {
    fn cmp(&self, other: &Self) -> cmp::Ordering {
        self.start.cmp(&other.start)
    }
}
```
The `cmp::Ordering` is enum including `Less`, `Equal`, `Greater`.
So that `==` and `>` operators of this class may be implemented.

Reply:
```rust
column_allocations
    .get_mut(c)
    .unwrap()
    .0
    .insert(AllocatedRegion {
        start: row,
        length: region_length,
     });
```
The `Ord` trait impl for `AllocatedRegion` is need for insertion to `BTreeSet`, so I think implementation of `operator<` is enough for now.